### ISSUE TITLE
🐛 Bug Report: Watsonx hardcoded values

### ISSUE THREAD

**Issue by gauthiersiri** [traceloop/openllmetry#2163]
### Which component is this bug for?

Watsonx Instrumentation

### 📜 Description

Some variables, showing in traceloop traces, are hardcoded.
For example, even if hitting watsonx in frankfurt, it shows that it's using https://us-south.ml.cloud.ibm.com, which is incorrect.

### 👟 Reproduction steps

Create a python script using watsonx in frankfurt for example, and see that in the traces it shows https://us-south.ml.cloud.ibm.com, as hardcoded in packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py line 103

### 👍 Expected behavior

It should show the real parameter istead of harcoded ones.

### 👎 Actual Behavior with Screenshots

![image](https://github.com/user-attachments/assets/f732128c-4848-4609-b721-56a40c2cd1e5)

### 🤖 Python Version

3.12

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

**Comment by nirga** [comment]
@gyliu513 fyi

**Comment by gyliu513** [comment]
Thanks @nirga @gauthiersiri , that is in our backlog, will fix it soon!

**Comment by rajmanna-dev** [comment]
Is this issue up to grab?

**Comment by gyliu513** [comment]
@rajmanna-dev Yes, as IBM Watsonx AI Platform also support on-prem as well, so with on-prem, the API BASE URL may changed.

**Comment by rajmanna-dev** [comment]
Can I take this issue?

**Comment by nirga** [comment]
Yes :) @rajmanna-dev

**Comment by rajmanna-dev** [comment]
Can you please assign me? @nirga

**Comment by gyliu513** [comment]
@rajmanna-dev just a fix proposal here, as the URL was mainly used as a span attribute, so we can probably let customer set a env var before they run their apps, and then we can update the code to use `os.getenv` to get the base URL, similar as [here](https://github.com/traceloop/openllmetry/blob/main/packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py#L117), this should works, good luck, look forward to your PR.

**Comment by nirga** [comment]
@gyliu513 you can't just get it from the client sdk?

**Comment by gyliu513** [comment]
@nirga No, from https://github.com/traceloop/openllmetry/blob/main/packages/sample-app/sample_app/watsonx-langchain.py#L31 , you can see this is configured by customer when they are trying to build the model for watsonx platform.

**Comment by nirga** [comment]
Ah great so we can get it from there!

**Comment by rajmanna-dev** [comment]
`packages/sample-app/sample_app/watsonx_generate.py` and `packages/sample-app/sample_app/watsonx_flow.py` also have the hardcoded api url values.

**Comment by nirga** [comment]
@rajmanna-dev that's ok - these are samples
